desktop/core/src/desktop/js/reactComponents/TaskBrowser/TaskBrowser.tsx

```diff
@@ -0,0 +1,434 @@
+          open={showSchedulePopup}
+        />
+      )}
+      <div style={{ margin: '20px 0' }}>
+        <TaskBrowserTable tasks={scheduledTasks} handleShowLogs={handleShowLogs} handleSchedulePopup={handleSchedulePopup}/>
+      </div>
+      {showLogPopup && (
+        <div className="popupStyle">
+          <div className="popupContentStyle">
+            <h2>Task Logs - {currentTaskId}</h2>
+            <pre>{taskLogs}</pre>
+            <button onClick={() => setShowLogPopup(false)}>Close</button>
+          </div>
+        </div>
+      )}
+    </div>
+
+  );
+};
+
+
+// A helper function to calculate duration
+const calculateDuration = (start, end) => {
+  if (!start || !end) return 'N/A'; // If either start or end is missing
+
+  const startDate = new Date(start);
+  const endDate = new Date(end);
+  const duration = endDate - startDate; // Duration in milliseconds
+
+  if (isNaN(duration)) return 'Invalid Dates'; // If dates can't be parsed
+
+  // Convert duration to hours, minutes, and seconds
+  let seconds = Math.floor((duration / 1000) % 60);
+  let minutes = Math.floor((duration / (1000 * 60)) % 60);
+  let hours = Math.floor((duration / (1000 * 60 * 60)) % 24);
+
+  // Build duration string
+  let durationStr = '';
+  if (hours > 0) durationStr += hours + 'h, ';
+  if (minutes > 0 || hours > 0) durationStr += minutes + 'm, ';
+  durationStr += seconds + 's';
+
+  return durationStr;
+};
+
+function formatTimestamp(timestamp) {
+  const options = { year: 'numeric', month: 'long', day: 'numeric', hour: '2-digit', minute: '2-digit', hour12: true };
+  const date = new Date(timestamp);
+  return date.toLocaleString('en-US', options);
+}
+
+
+export const TaskBrowserTable = ({ handleShowLogs, handleSchedulePopup }) => {
+
+  const [tasks, setTasks] = useState([]);
+  const [searchTerm, setSearchTerm] = useState(''); // state for the search term
+  const [statusFilter, setStatusFilter] = useState({ success: false, failure: false, running: false, all: true }); // state to track status filter
+  const [selectedTasks, setSelectedTasks] = useState([]); // state for check box of a task
+
+  useEffect(() => {
+    const fetchTasks = () => {
+    axios.get('/desktop/api2/get_taskserver_tasks/')
+        .then(response => {
+            if (Array.isArray(response.data)) {
+            //sort based on timestamp
+            const sortedTasks = response.data.sort((a, b) => {
+                // Assuming 'date_done' is the timestamp field and is in a comparable format
+                return new Date(b.date_done) - new Date(a.date_done);
+            });
+            setTasks(sortedTasks);
+          } else {
+            console.error('Expected an array of tasks, but received:', response.data);
+          }
+        })
+        .catch(error => {
+            console.error('Error fetching tasks', error);
+        });
+    };
+
+    fetchTasks();   //fetch tasks initially
+    const interval = setInterval(fetchTasks, 5000);
```

Comment:
Also here I think it's a better pattern that fetchTasks triggers a new fetchTasks after 5 seconds with a timeout instead of an interval.